packages/cookies/src/request-cookies.ts

```diff
@@ -1,80 +1,106 @@
-import { parseCookieString, serialize } from './serialize'
+import { type Cookie, parseCookieString, serialize } from './serialize'
 import { cached } from './cached'
 
 /**
  * A class for manipulating {@link Request} cookies.
  */
 export class RequestCookies {
-  private readonly headers: Headers
+  readonly #headers: Headers
 
   constructor(request: Request) {
-    this.headers = request.headers
+    this.#headers = request.headers
   }
 
-  /**
-   * Delete all the cookies in the cookies in the request
-   */
-  clear(): void {
-    this.delete([...this.parsed().keys()])
+  #cache = cached((header: string | null) => {
+    const parsed = header ? parseCookieString(header) : new Map()
+    return parsed
+  })
+
+  #parsed(): Map<string, string> {
+    const header = this.#headers.get('cookie')
+    return this.#cache(header)
   }
 
-  /**
-   * Format the cookies in the request as a string for logging
-   */
-  [Symbol.for('edge-runtime.inspect.custom')]() {
-    return `RequestCookies ${JSON.stringify(Object.fromEntries(this.parsed()))}`
+  [Symbol.iterator]() {
+    return this.#parsed()[Symbol.iterator]()
   }
 
   /**
    * The amount of cookies received from the client
    */
   get size(): number {
-    return this.parsed().size
+    return this.#parsed().size
   }
 
-  [Symbol.iterator]() {
-    return this.parsed()[Symbol.iterator]()
+  get(...args: [name: string] | [Cookie]) {
+    const name = typeof args[0] === 'string' ? args[0] : args[0].name
+    return this.#parsed().get(name)
   }
 
-  private cache = cached((header: string | null) => {
-    const parsed = header ? parseCookieString(header) : new Map()
-    return parsed
-  })
-
-  private parsed(): Map<string, string> {
-    const header = this.headers.get('cookie')
-    return this.cache(header)
-  }
+  getAll(...args: [name: string] | [Cookie] | [undefined]) {
+    const all = Array.from(this.#parsed())
+    if (!args.length) {
```

Comment:
Maybe a strict comparison could be desirable

```
args.length === 0
```

Reply:
Because of TypeScript, this would show:
> This condition will always return 'false' since the types '1' and '0' have no overlap.ts(2367)

Since we technically don't allow more than one argument, but the user might/will try.